### PR TITLE
NPUW: Support Qwen3 TTS talker language model

### PR DESCRIPTION
**Pull request by RyanMetcalfeInt8**
### Details:
This PR adds some necessary changes to support Qwen3 TTS 'talker' language model with NPUW's existing LLM flow (i.e. "NPUW_LLM": "YES")

This model has a slightly different signature than other models which are currently supported by this flow:
```
        Inputs:
                attention_mask, i64, [?,?], [...]
                position_ids, i64, [3,?,?], [...]
                inputs_embeds, f32, [?,?,1024], [...]
                beam_idx, i32, [?], [...]
        Outputs:
                logits, f32, [?,?,3072], [...]
                hidden_states, f32, [?,?,1024], [...] 
```

Notice that hidden_states is also connected as output tensor – as hidden_states are needed (along with logits of course) to pass onto code predictor stage of Qwen3 TTS pipeline.

Initially, a crash was observed and was root-caused to a thrown exception causing a use-after-free. So this first commit addresses that crash: https://github.com/openvinotoolkit/openvino/commit/0a0f013bec8f805ae2682a2f4277dcfc37565f00 

After fixing that crash, the 'real' exceptions were exposed:
```
in exception: Exception from src\inference\src\cpp\core.cpp:120:
Exception from src\inference\src\dev\plugin.cpp:54:
Exception from src\plugins\intel_npu\src\plugin\npuw\compiled_model.cpp:1629:
NPUW: Assertion from_submodel != NO_LINK failed
```

And so the following commits start to address those issues. 

### Tickets:
 - EISW-224187

### AI Assistance:
 - *AI assistance used: yes*
 - *If yes, summarize how AI was used and what human validation was performed (build/tests/manual checks).*
 AI was used to analyze the various issues (crashes, exceptions) when attempting to compile Qwen3 talker language model with NPUW. It also was used to analyze existing NPUW routines (as I am not an expert here) and suggest / apply fixes.
 
 I verified that, after these fixes, that I am able to compile & run this model using NPUW when dropped into the E2E pipeline, and that performance / accuracy look good.

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp [hunk 1]
```diff
@@ -735,17 +760,31 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
...
-        LOG_BLOCK();
-        log_device_dist();
-    }
+    // finalize_weights_bank() launches an asynchronous weights-evaluation task that
+    // captures this CompiledModel's members by reference (see m_eval_future). The
+    // destructor joins that task, but a C++ destructor is NOT invoked when an
+    // exception escapes the constructor - so if anything below throws, the async
+    // task would keep running against members that are about to be destroyed,
+    // causing a use-after-free. Guard the remainder of construction: on failure,
+    // wait for the async evaluation to complete before letting the exception out.
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:814]: The whole point of this future thing was to "catch" the synchronization point in the infer request constructor - in our normal flow we get to it after some time as there's multiple models compiled in the pipeline.

So we intentionally don't wait for the `future` right here - but, probably, we should wait for one in the destructor if it hits first (and the infer request is not created)

**Review by dmatveev — COMMENTED** [review]
@AlexanderKalistratov @AsyaPronina can you please review the rest of the changes

**Comment by RyanMetcalfeInt8** [comment]
Just realized there are some issues when model is imported as blob (e.g. cache_dir scenario)-- I think due to some logic that is skipped in default LLMCompiledModel constructor. Let me fix / test that, and I'll open it back up for review.

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp (hunk 1 above)
Comment (RyanMetcalfeInt8) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:814]: > So we intentionally don't wait for the future right here - but, probably, we should wait for one in the destructor if it hits first (and the infer request is not created)

Right, it's a little bit fuzzy to me, but I think the main issue was that an exception was thrown by one of these functions kicked off during this constructor, and so destructor isn't called (which is where the async tasks were synchronized, and those tasks used members of this class which got destroyed).

**Comment by AlexanderKalistratov** [comment]
Every time you update serialization/deserialization functions you has to bump [NPUW_SERIALIZATION_VERSION ](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_npu/src/plugin/npuw/serialization.hpp#L51)

**Comment by RyanMetcalfeInt8** [comment]
> Every time you update serialization/deserialization functions you has to bump [NPUW_SERIALIZATION_VERSION ](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_npu/src/plugin/npuw/serialization.hpp#L51)

Thanks, yes I had originally bumped it up to 0.28, but looks like after merge from master it needed to get bumped again. Anyway, done.

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 2]
```diff
@@ -1318,7 +1378,7 @@ void ov::npuw::LLMCompiledModel::serialize(std::ostream& raw_stream, const ov::n
             m_kvcache_desc.v_tensors_transposed_gen & m_prefill_chunk_size & m_use_chunk_prefill & m_max_lora_rank &
             m_enable_prefix_caching & m_prefix_caching_block_size & m_prefix_caching_max_num_blocks &
             m_longrope_context_limit & m_is_whisper & m_eos_token_id & m_decomposed_sdpa_size & m_is_eagle &
-            m_is_embedding & m_is_block_kv_cache;
+            m_is_embedding & m_is_block_kv_cache & m_has_lm_head_hidden_states;
```
[the remark this answers is not in the document]
Comment (RyanMetcalfeInt8) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp]: fixed, bumped NPUW_SERIALIZATION_VERSION .

**Comment by AsyaPronina** [comment]
Hi @RyanMetcalfeInt8 , I believe we need to have a call to merge two of our PRs together: https://github.com/openvinotoolkit/openvino/pull/33664

**Comment by dmatveev** [comment]
@RyanMetcalfeInt8 can you please specify which release this PR targets to?

**Comment by RyanMetcalfeInt8** [comment]
> @RyanMetcalfeInt8 can you please specify which release this PR targets to?

@dmatveev, as far as I know the plan is to support Qwen3 TTS for 26.4. Draft PR for GenAI is here: https://github.com/openvinotoolkit/openvino.genai/pull/4189

But currently awaiting optimum-intel support.

**Comment by dmatveev** [comment]
@RyanMetcalfeInt8 thanks. Please ignore @AsyaPronina 's comment then, that PR won't make it in time - lets agree on the required changes only in scope of yours.

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 3]
```diff
@@ -132,6 +132,17 @@ class CutLMHead : public ov::pass::MatcherPass {
...
+            // NOTE: matmul_first_source is the model's final hidden-state tensor. Some
+            //       models also expose that same tensor through a separate Result (e.g. a
+            //       "hidden_states" output). set_names() replaces the source's name set,
+            //       so after the cut any such sibling Result and matched_result both read
+            //       matmul_first_source and carry the identical "npuw_output_embed" name.
+            //       Runtime port maps (unordered_map, keyed by name) would then silently
+            //       drop one of the two same-named outputs, leaving the lm_head input
+            //       wired to a dead/constant tensor and freezing decode. cut_lm_head()
+            //       removes redundant sibling Results right after this pass so exactly
+            //       one live output survives.
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:145]: @RyanMetcalfeInt8 can you please confirm if the two (or more) Results attached to the model's output port, that are handled in `compiled_model.cpp`, are coming from  this situation? In fact, the extra output attached is the one done by CutLMHead?

**Review by dmatveev — COMMENTED** [review]
Self-note: All the changes in this PR are caused by the fact that the model seem to have one more output - right before the LMHead cut point. It shouldn't have specific handling, needs to be solved in a generic way instead (the current code assumes for LLMCompiledModel there's just a single output possible)

File: src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp [hunk 4]
```diff
@@ -1707,6 +1708,20 @@ ov::SoPtr<ov::ITensor> ov::npuw::LLMInferRequest::get_tensor(const ov::Output<co
         }
     }
 
+    // Models that expose a "hidden_states" output alongside "logits" (e.g. the
+    // Qwen3-TTS talker) have that Result removed from the internal submodel during
+    // cut_lm_head(). At runtime the embed tensor shared between the
+    // prefill/generate model and the lm_head input holds the last valid token's
+    // hidden state - return it here.
+    if (m_npuw_llm_compiled_model->m_has_lm_head_hidden_states && m_lm_head_request &&
+        port_names.count("hidden_states") > 0) {
```
Comment (RyanMetcalfeInt8) [inline src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp:1717]: This also needs to be addressed -- new exported model from updated optimum-intel PR has output tensor named `"last_hidden_state"`
